### PR TITLE
Example building a docker image with Kaniko

### PR DESCRIPTION
**Pull request by carlossg**

### CODE REVIEW

File: examples/kaniko.groovy [hunk 1]
```diff
@@ -0,0 +1,43 @@
+/**
+ * This pipeline will build and deploy a Docker image with Kaniko
+ * https://github.com/GoogleContainerTools/kaniko
+ * without needing a Docker host
+ *
+ * You need to create a jenkins-docker-cfg secret with your docker config
+ * as described in
+ * https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/#create-a-secret-in-the-cluster-that-holds-your-authorization-token
```
Comment (kmadel) [inline examples/kaniko.groovy]: Those instructions are specifically for pulling images used by a `pod` - not for pushing to a registry. I had to use the following `secret` configuration to make sure the Docker credentials were placed in `~/.docker/config.json`:

```
...
     volumeMounts:
       - name: jenkins-docker-cfg
         mountPath: /root/.docker
   volumes:
     - name: jenkins-docker-cfg
       secret:
         secretName: jenkins-docker-cfg
         items:
         - key: .dockerconfigjson
           path: config.json
```

Comment (carlossg) [reply examples/kaniko.groovy]: I haven't tried, but I assumed it was the same format, will take a look.

This I have tried https://github.com/GoogleContainerTools/kaniko/pull/129

Comment (carlossg) [reply examples/kaniko.groovy]: Figured it out, the docker-registry secret type crates a .dockercfg file that needs to be put in root
It does not create the new .docker/config.json format

File: examples/kaniko.groovy [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+ def label = "kaniko-${UUID.randomUUID().toString()}"
+
+ podTemplate(name: 'kaniko', label: label, yaml: """
+ kind: Pod
+ metadata:
+   name: kaniko
+ spec:
+   containers:
+   - name: kaniko
+     image: csanchez/kaniko:jenkins # we need a patched version of kaniko for now
```
Comment (StevenACoffman) [inline examples/kaniko.groovy]: Is this because of GoogleContainerTools/kaniko#129 or something else?

Comment (carlossg) [reply examples/kaniko.groovy]: No, need cat and nohup in the image https://github.com/carlossg/kaniko/blob/jenkins/deploy/Dockerfile_jenkins

**Comment by kmadel** [comment]
Are you sure that `nohup` is in your image. Seeing this:
```
[testKanikoExample] Running shell script
/bin/sh: nohup: not found
EXITCODE 127[Pipeline] }
```

**Comment by StevenACoffman** [comment]
[coreutils contents](https://pkgs.alpinelinux.org/contents?repo=main&page=2&arch=armhf&branch=edge&name=coreutils) show it there, what sha256 do you have?

**Comment by carlossg** [comment]
fixed and image is building in dockerhub

**Pull request merged**